zilliqa/src/consensus.rs

```diff
@@ -1503,6 +1503,33 @@ impl Consensus {
             }
         }
 
+        if block.number() % self.config.consensus.blocks_per_epoch == 0 && block.number() != 0 {
+            // TODO: handle epochs
+            if self.config.do_snapshots
+                && block.number()
+                    % (self.config.consensus.blocks_per_epoch
+                        * self.config.consensus.epochs_per_checkpoint)
+                    == 0
+            {
+                if let Some(snapshot_path) = self.db.create_checkpoint_path(block.number())? {
```

Comment:
I've now added old checkpoing cleanup - since you mention it, I don't see a good reason to keep _any_ old checkpoints, so I just clean the entire directory every time the new one is saved (after it's successfully saved, of course, so we always have at least one valid checkpoint!).